src/edit/components/PrimaryToggleButton/PrimaryToggleButton.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Primary Toggle Button
+ */
+
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { change } from 'redux-form';
+import { Button, Label } from '@folio/stripes/components';
+import { FormattedMessage } from 'react-intl';
+
+const PrimaryToggleButton = ({ label, meta: { dispatch, form }, input: { value, onChange }, fields }) => {
+  const isPrimary = value === true;
+  const handleChange = () => {
+    if (isPrimary) {
+      return;
+    }
+
+    // Reset other primary fields
+    fields.forEach(fieldName => dispatch(change(form, `${fieldName}.primary`, false)));
+
+    // Set primary flag for current field
+    onChange(true);
+  };
+
+  return (
+    <Fragment>
+      { label && <Label>{label}</Label>}
+      <Button
+        data-test-primary-toggle-button
+        buttonStyle={isPrimary ? 'primary' : 'default'}
+        onClick={handleChange}
```

Comment:
I believe, `handleClick`, as a handler name, will look a little bit accurate. But it's up to you :)

Reply:
Makes more sense, yes. I'll change it. Thanks :)